src/main/java/com/shaft/support/DBActions.java

```diff
@@ -0,0 +1,155 @@
+package com.shaft.support;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+import org.testng.Assert;
+
+import com.shaft.io.ReportManager;
+
+public class DBActions {
+
+	Connection connection;
+	Statement statement;
+	ResultSet resultSet;
+	String dbType;
+	String dbServerIP;
+	String dbPort;
+	String dbName;
+	String username;
+	String password;
+
+	public DBActions(String dbType, String dbServerIP, String dbPort, String dbName, String username, String password) {
+		this.dbType = dbType;
+		this.dbServerIP = dbServerIP;
+		this.dbPort = dbPort;
+		this.dbName = dbName;
+		this.username = username;
+		this.password = password;
+	}
+
+	private void passAction(String actionName, String testData, String log) {
```

Comment:
this method is the one that should be called after the query runs successfully, and the result string from the query execution should be passed to the log object.

Reply:
It is added in executeSelectQuery but I passed the query for it instead of result because in this layer the result is object of ResultSet not string, thoughts ?

Reply:
in executeSelectQuery, the method used is this one [passAction("Execute DB query", dbQuery);]

I'm asking you to use this one instead [passAction("Execute DB query", dbQuery, resultSet);]

There's another comment advising that the returned result set should be parsed to string and not returned as is. In which case you can parse it, report that the step has passed, and then return the string.

Reply:
will reopen comment on the latest commit.